boot-core/src/daemon/contract.rs

```diff
@@ -7,31 +7,45 @@ use std::{
     env,
     fmt::{self, Debug},
     fs,
-    path::Path,
+    path::Path
 };
 
 impl Contract<Daemon> {
     /// Only upload the contract if it is not uploaded yet (checksum does not match)
-    /// @TODO proper response
     pub fn upload_if_needed(&mut self) -> Result<Option<TxResponse<Daemon>>, BootError> {
         if self.latest_is_uploaded()? {
             log::info!("{} is already uploaded", self.id);
             Ok(None)
         } else {
+            log::info!("{} is not uploaded, uploading...", self.id);
             Some(self.upload()).transpose()
         }
     }
 
-    /// Returns a bool whether the checksum of the wasm file matches the checksum of the previously uploaded code
+    // Returns a bool whether the checksum of the wasm file matches the checksum of the previously uploaded code
```

Comment:
Yeah, while this probably will not fail, any network request could still fail and if it does then letting that error propagate up helps the dev identify the issue (his wifi going down/whatever). 

Also in a horror scenario the `latest_uploaded_code_id` could be completely wrong due to a dev using the wrong state file for example, which could also generate an error (like that the code-id doesn't exist)

Reply:
Ok I see your point.

the function above `upload_if_needed` is not working as intended , because it's getting this error:

`upload_if_needed response: Err(
    DaemonError(
        Status(
            Status {
                code: Unknown,
                message: "not found",
                metadata: MetadataMap {
                    headers: {
                        "content-type": "application/grpc",
                        "x-cosmos-block-height": "1",
                    },
                },
                source: None,
            },
        ),
    ),
)`

im reversing the code to its original form and await more feedback on this